mono/metadata/marshal.c

```diff
@@ -3621,9 +3621,12 @@ mono_marshal_get_delegate_invoke_internal (MonoMethod *method, gboolean callvirt
 			mono_mb_emit_ldloc (mb, local_target);
 			for (i = 0; i < sig->param_count; ++i)
 				mono_mb_emit_ldarg (mb, i + 1);
-			mono_mb_emit_ldarg (mb, 0);
-			mono_mb_emit_ldflda (mb, MONO_STRUCT_OFFSET (MonoDelegate, extra_arg));
-			mono_mb_emit_byte (mb, CEE_LDIND_I);
+
```

Comment:
not exactly sure how the JIT handles it, but indeed, we can just remove the argument from the stack when we see a `MONO_CALLI_EXTRA_ARG`: https://github.com/mono/mono/pull/4430/commits/c66806a5ad54db6e973ffa649a36028ea153866c

thus, I removed the changes to `marshal.c` 👍